instance/management/commands/update_metadata.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015-2018 OpenCraft <contact@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Instance app - Activity report management command
+"""
+
+# Imports #####################################################################
+
+import sys
+
+from django.core.management.base import BaseCommand
+
+from instance.models.openedx_instance import OpenEdXInstance
+
+
+# Classes #####################################################################
+
+
+class Command(BaseCommand):
+    """
+    management command class for updating instances metadata
+    """
+    help = 'Updates the Consul metadata for all instances.'
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '-i',
+            '--instance',
+            type=int,
+            required=False,
+            help='Specific Instance ID you want to update its metadata.'
+        )
+
+    def handle(self, *args, **options):
+        instance_id = options.get('instance')
+
+        if instance_id:
+            self.update_instance_metadata(instance_id)
+        else:
+            self.update_all_instances_metadata()
+
+    def update_instance_metadata(self, instance_id):
+        self.stdout.write('Updating instance {}\' metadata...'.format(
+            instance_id))
+
+        try:
+            instance = OpenEdXInstance.objects.get(id=instance_id)
+        except:
+            self.stdout.write(self.style.ERROR(
+                'Couldn\'t find an instance with id: {}'.format(
+                    instance_id)))
+            sys.exit(1)
+
+        instance.update_consul_metadata()
+        self.stdout.write(self.style.SUCCESS(
+                'Successfully update instances {}\' metadata'.format(
+                    instance_id)))
+
+    def update_all_instances_metadata(self):
+        self.stdout.write('Updating all instances\' metadata...')
+        for instance in OpenEdXInstance.objects.all():
+            instance.update_consul_metadata()
```

Comment:
This does not remove archived instances from the list in Consul – see my comment on the ticket for more details.